management/appointments/appointmentsController.js

```diff
@@ -53,39 +68,44 @@ define('appointmentsController', [], function() {
 					$scope.submittingReschedule = false;
 					
 					// Let the user know it was successful
-					WDN.initializePlugin('notice');
-					var body = angular.element( document.querySelector( 'body' ) );
-					body.append(`
-						<div class="wdn_notice affirm" data-overlay="maincontent" data-duration="10">
-							<div class="close">
-								<a href="#">Close this notice</a>
-							</div>
-							<div class="message">
-								<p class="title">Success!</p>
-								<p>This appointment was successfully rescheduled.</a>
-								</p>
-							</div>
-						</div>`);  
+					$scope.giveNotice("Success!", "This appointment was successfully rescheduled.", true);
 				} else {
 					alert(result.error);
 
 					$scope.submittingReschedule = false;
 
 					// Let the user know it failed
-					WDN.initializePlugin('notice');
-					var body = angular.element( document.querySelector( 'body' ) );
-					body.append(`
-						<div class="wdn_notice negate" data-overlay="maincontent" data-duration="10">
-							<div class="close">
-								<a href="#">Close this notice</a>
-							</div>
-							<div class="message">
-								<p class="title">Failure</p>
-								<p>Something went wrong and this appointment was not rescheduled!</a>
-								</p>
-							</div>
-						</div>`);
+					$scope.giveNotice("Failure", "Something went wrong and this appointment was not rescheduled!", false);
+				}
+			});
+		}
+
+		$scope.cancelAppointment = function() {
+			if ($scope.cancelling) return false;
+
+			$scope.cancelling = true;
+
+			const appointmentId = $scope.appointment.appointmentId;
+			AppointmentsService.cancelAppointment(appointmentId).then(function(result) {
+				if (result.success) {
+					document.body.scrollTop = document.documentElement.scrollTop = 0;
+
+					$scope.appointment.cancelled = true;
+					$scope.appointment.notStarted = false;
+					$scope.appointment.statusText = "Cancelled";
+					$scope.appointment.notCompletedDescription = "Cancelled Appointment";
+
+					// Let the user know it was successful
+					$scope.giveNotice("Success!", "This appointment was successfully cancelled.", true);
```

Comment:
The last argument is not necessary because you stipulate `true` is the default argument value.

Reply:
True, I always enjoy being more explicit, but the more I think about it, I think not specifying when I give a default is significantly better.